### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by dipshojol**

### CODE REVIEW

**Review by Badr-KH — CHANGES_REQUESTED** [review]
I think the Dialog.js file is better suited to be in the component folder rather than utils. Concerning your navbar code, I see that you made every link open a dialog rather than redirect to the appropriate page. If this is the case, will we have to design a different dialog for every page?

File: client/src/Components/Nav.js [hunk 1]
```diff
@@ -57,12 +64,12 @@ const Nav = () => {
                     variant="outlined"
                     />
                 <ul className={classes.ul}>
-                    <NavLink className={classes.a} to="/profile" activeClassName="active"><li className={classes.li}>Share music</li></NavLink>
-                    <NavLink className={classes.a} to="/discover" activeClassName="active"><li className={classes.li}>Discover</li></NavLink>
-                    <NavLink className={classes.a} to="/messages" activeClassName="active"><li className={classes.li}>Messages</li></NavLink>
+                    <a href="./" className={classes.a}><li className={classes.li}><NavDialog label={"Share Music"} /></li></a>
+                    <a href="./" className={classes.a}><li className={classes.li}><NavDialog label={"Discover"} /></li></a>
+                    <a href="./" className={classes.a}><li className={classes.li}><NavDialog label={"Massages"} /></li></a>
```
Comment (aeciorc) [inline client/src/Components/Nav.js]: I think you still want Discover and Messages to be routes. And for the ShareMusic dialog, you want it to be a button, not an anchor tag. Maybe change the placement to make it more clear that Share Music is different then the other items on the nav

Comment (aeciorc) [reply client/src/Components/Nav.js]: Also, does the dialog actually open? Usually, the logic for closing/opening is in the parent component and `open` is passed as a prop

File: client/src/pages/LoginForm.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React, { useState } from "react";
-import useStyles from "../styles/formstyles";
```
Comment (aeciorc) [inline client/src/pages/LoginForm.js]: This shouldn't have been renamed, same as below

**Comment by dipshojol** [comment]
> I think the Dialog.js file is better suited to be in the component folder rather than utils. Concerning your navbar code, I see that you made every link open a dialog rather than redirect to the appropriate page. If this is the case, will we have to design a different dialog for every page?

in the nav bar, as of now we dont have any plan for other tabs like messages and discover.

File: client/src/Components/Nav.js (hunk 1 above)
Comment (dipshojol) [reply client/src/Components/Nav.js]: yaa. please check now.

File: client/src/pages/LoginForm.js (hunk 2 above)
Comment (dipshojol) [reply client/src/pages/LoginForm.js]: fixed

**Review by aeciorc — COMMENTED** [review]
I tested this today. Found some issues:

1) It's complaining about formstyles not being found
2) Discover and Message links are still not taking user to those routes
3) The dialog still isn't properly setup with the setter in the parent component. If you're not sure how to do that, let me know and I'll help implement it

**Comment by dipshojol** [comment]
> I tested this today. Found some issues:
> 
> 1. It's complaining about formstyles not being found
> 2. Discover and Message links are still not taking user to those routes
> 3. The dialog still isn't properly setup with the setter in the parent component. If you're not sure how to do that, let me know and I'll help implement it

pushed the changes according to the requirements. please check

**Comment by dipshojol** [comment]
@aeciorc 

> I think the Dialog.js file is better suited to be in the component folder rather than utils. Concerning your navbar code, I see that you made every link open a dialog rather than redirect to the appropriate page. If this is the case, will we have to design a different dialog for every page?

Hi @Badr-KH , moved the dialog box to components. could you please review it again so that can be merge..

**Comment by dipshojol** [comment]
@aeciorc could you please check the pull request now.

**Comment by dipshojol** [comment]
> I think the Dialog.js file is better suited to be in the component folder rather than utils. Concerning your navbar code, I see that you made every link open a dialog rather than redirect to the appropriate page. If this is the case, will we have to design a different dialog for every page?

done

File: client/src/Components/Dialog.js [hunk 3]
```diff
@@ -33,11 +24,10 @@ export default function NavDialog({ label }) {
 				placeSelf: 'center'
 			}}
 		>
-			<Button onClick={handleClickOpen}>{label}</Button>
 			<Dialog
 				fullScreen={fullScreen}
 				open={open}
-				onClose={handleClose}
+				onClose={event => close(false)}
```
Comment (aeciorc) [inline client/src/Components/Dialog.js]: `close ` takes no arguments, so this should be  `onClose={close}`

**Review by aeciorc — APPROVED** [review]
Approved with a comment below

File: client/src/Components/Nav.js [hunk 4]
```diff
@@ -1,71 +1,105 @@
...
+				<Button className={classes.dialogButton} onClick={openDialog}>Share Music</Button>
+				<ul className={classes.ul}>
+					<a href='./discover' className={classes.a}>
+						<li className={classes.li}>
+							Discover
+						</li>
+					</a>
+					<a href='./messages' className={classes.a}>
+						<li className={classes.li}>
+							Massages
```
Comment (aeciorc) [inline client/src/Components/Nav.js]: Messages* 😃

File: client/src/Components/Dialog.js (hunk 3 above)
Comment (dipshojol) [reply client/src/Components/Dialog.js]: done

File: client/src/Components/Nav.js (hunk 4 above)
Comment (dipshojol) [reply client/src/Components/Nav.js]: done

File: client/src/Components/Dialog.js [hunk 5]
```diff
@@ -14,6 +14,10 @@ export default function NavDialog({ open, close }) {
 	const theme = useTheme();
 	const fullScreen = useMediaQuery(theme.breakpoints.down('sm'));
 
+	const closeDialog = () => {
+		return close(false)
+	}
+
```
Comment (aeciorc) [inline client/src/Components/Dialog.js]: I don't think you need this though, cause `closeDialog `is already defined as `setOpen(false)`

Comment (dipshojol) [reply client/src/Components/Dialog.js]: yaa, right. its ok now

**Comment by aeciorc** [comment]
Ok, looks good now

**Review by Badr-KH — APPROVED** [review]

**Pull request merged**